README.md

```diff
@@ -7,51 +7,53 @@ This project was built using only built in web services (ie no NPM packages besi
+
+The framework being used caused for some hiccups when getting started with this API. Lit is a web components frame work where you can create these smaller building block components and bring them together to create a full webpage and when nested create `shadowRoots` that have thier own scope (per component). This causes functions that are normally called like `window.getSelection()` to not return what the documentation says you should expect because things are buried within shadowRoots. In chromium, `shadowRoot.getSelection()` is supported and due to this, the Notepad PWA will only work in chromium environments since find/replace and all the edit functions are core to the experience.
+
+### 🗄️ Local Storage
+
+This PWA uses local storage to save a lot of state variables and settings. This includes saving the last entered text into the editor (even if the user doesn't save) to make sure the PWA reopens to this text next time the launch the app, also ensuring their settings are persistent across mutiple sessions of using the app.
+
+We began the work of adding tabs to the Notepad PWA to get closer to the Native app, however, we ran out of time and wanted to prioritize discovery of the gap between Native and PWA rather than how well can we code a Notepad PWA. Introducing tabs adds a layer of complexity with using local storage because now there are somethings that need to persist across all of Notepad (Settings or the last thing the user searched) and others that are specific to the Notepad instance or tab (the text in the editor, the file handler).
+
+While using local storage is usually pretty light on processing power, if you are going to be reading and writing to local storage often, it can begin to slow down your App, though we never reached that point with the Notepad PWA.
 
 ### 👷🏾‍♂️ Service Worker
-- NOTEPAD PWA ALSO WORKS OFFLINE
-- Limitations:
-  - How tricky things get with tabs
+
+Service workers and PWAs go hand in hand with the most common use case being getting the app working (to some extent) offline. On first visit, the Notepad PWA Service worker will register and save all of the assets to the cache. After this, anytime the user loses connection the assets are saved so visually things work and look the same. The data entered while offline uses local storage which doens't need connection anyway so the editor would have worked offline by default.
+
+We gave our service worker the "cache first" stratergy which serves the user anything in the cache before anything else loaded from the apps server. This can create faster load times for your user, however, it can also cause things to become stale if the developer pushes new updates to their app. This is avoided with the addition of a [little bit of code](https://web.dev/articles/offline-cookbook#stale-while-revalidate).
```

Comment:
```suggestion
We built our service worker with the "cache first" strategy, which serves the user anything in the cache before anything else loaded from the app's server. This can create faster load times for the user. However, it can also cause things to become stale if the developer pushes new updates to their app. This is avoided with the addition of a [little bit of code](https://web.dev/articles/offline-cookbook#stale-while-revalidate).
```